src/main/java/eutros/multiblocktweaker/crafttweaker/construction/MultiblockBuilder.java

```diff
@@ -121,6 +127,49 @@ public MultiblockBuilder withZoom(float zoom) {
         return this;
     }
 
+    /**
+     * Add a tooltip to a specific {@link net.minecraft.item.ItemStack} in a Multiblock Structure.
+     * Multiple tooltips can be added to the same {@link net.minecraft.item.ItemStack}.
+     * This is optional, and if not defined will not add any tooltips.
+     *
+     * @param itemStack - The ItemStack form of the block to add the tooltip to.
+     * @param tooltip - The localization key for the tooltip to be added.
+     * @return This builder, for convenience.
+     */
+    @ZenMethod
+    public MultiblockBuilder withPartTooltip(IItemStack itemStack, crafttweaker.api.text.ITextComponent tooltip) {
+
+        ItemStack actualStack = CraftTweakerMC.getItemStack(itemStack);
+        ITextComponent tooltipText = CraftTweakerMC.getITextComponent(tooltip);
+
+        List<ITextComponent> possibleTooltips = tooltipMap.getOrDefault(actualStack, null);
+
+        if(possibleTooltips == null) {
+            List<net.minecraft.util.text.ITextComponent> tooltipToAdd = new ArrayList();
+            tooltipToAdd.add(tooltipText);
+            this.tooltipMap.put(actualStack, tooltipToAdd);
+        } else {
+            possibleTooltips.add(tooltipText);
+            this.tooltipMap.put(actualStack, possibleTooltips);
+        }
```

Comment:
Could this been done with
```java
tooltipMap.computeIfAbsent(actualStack, $ -> new ArrayList<>()).add(tooltipText);
```
?
The `put` in the `else` block is redundant anyway since the same exact key/value pair would already be in the map, and this would convey intent more clearly,

(Although with the identity hash map, the else statement is unlikely to get hit)

Reply:
`computeIfAbsent` would only handle half the case, because we also want to append the passed tooltip to the existing tooltip value for the passed key, which I am having trouble thinking of how to do that if `computeIfAbsent` is called first.